tiledb/sm/array/array.cc

```diff
@@ -1491,6 +1488,38 @@ Status Array::compute_max_buffer_sizes(
   return Status::Ok();
 }
 
+void Array::load_local_metadata() {
+  assert(array_dir_.loaded());
```

Comment:
This statement shouldn't use `assert`. Instead. failing the same check should be upgraded to `throw`. The whole loading regime with `class Array` is fragile; we should be checking in release builds if we ever needed to check in debug ones.

There are a few other occurrence of `assert` in `array.cc`. It looks like they should all be upgraded to `throw`. No need to do it in this PR, but keep an eye out for them.

Reply:
Switched to a `throw`.